File: 2025_01/leetcode2025-01-13.py
from typing import List

class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        tot = sum(nums)
        pre = 0
        res = 0
        for i in nums[:-1]:
            pre += i
            if 2*pre >= tot:
                res += 1
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [10,4,-8,7]
    print(s.waysToSplitArray(nums))

File: leetcode2025/2025_01/leetcode2025-01-13.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        long long tot = 0;
        for (int i : nums){
            tot += i;
        }
        long long pre = 0, res = 0;
        for (int i=0; i<nums.size()-1; i++){
            pre += nums[i];
            if (2*pre >= tot){
                res++;
            }
        }
        return res;
    }
}; 

int main(){
    Solution s;
    vector<int> nums = {2,3,1,0};
    cout << s.waysToSplitArray(nums) << endl;
}
